### ISSUE TITLE
An error occurred while installing digest-crc (0.6.1), and Bundler cannot continue. 

### ISSUE THREAD

**Issue by iadeelzafar** [fastlane/fastlane#17399]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I'm trying to Install Fastlane on Ubuntu via Gemfile

Here's what my Gemfile looks like
```
source "https://rubygems.org"

gem "fastlane"
gem "rake"
```
When running, `sudo bundle update`, I get the following error

```
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 13.0.1
Following files may not be writable, so sudo is needed:
  /var/lib/gems/2.7.0
  /var/lib/gems/2.7.0/build_info
  /var/lib/gems/2.7.0/cache
  /var/lib/gems/2.7.0/doc
  /var/lib/gems/2.7.0/extensions
  /var/lib/gems/2.7.0/gems
  /var/lib/gems/2.7.0/specifications
Using CFPropertyList 3.0.2
Using public_suffix 4.0.6
Using addressable 2.7.0
Using atomos 0.1.3
Using aws-eventstream 1.1.0
Using aws-partitions 1.380.0
Using aws-sigv4 1.2.2
Using jmespath 1.4.0
Using aws-sdk-core 3.109.1
Using aws-sdk-kms 1.39.0
Using aws-sdk-s3 1.83.0
Using babosa 1.0.4
Using bundler 2.1.2
Using claide 1.0.3
Using colored 1.2
Using colored2 3.1.2
Using highline 1.7.10
Using commander-fastlane 4.4.6
Using declarative 0.0.20
Using declarative-option 0.1.0
Fetching digest-crc 0.6.1


Your user account isn't allowed to install to the system RubyGems.
  You can cancel this installation and run:

      bundle install --path vendor/bundle

  to install the gems into ./vendor/bundle/, or you can enter your password
  and install the bundled gems to RubyGems using sudo.

  Password: 
Installing digest-crc 0.6.1 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/gems/digest-crc-0.6.1/ext/digest
/usr/bin/ruby2.7 -I/usr/lib/ruby/2.7.0/rubygems -rrubygems /usr/share/rubygems-integration/all/gems/rake-13.0.1/exe/rake
RUBYARCHDIR\=/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/extensions/x86_64-linux/2.7.0/digest-crc-0.6.1
RUBYLIBDIR\=/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/extensions/x86_64-linux/2.7.0/digest-crc-0.6.1
/usr/bin/ruby2.7 -S extconf.rb
checking for stdint.h... *** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/usr/bin/$(RUBY_BASE_NAME)2.7
	--with-stdint-dir
	--without-stdint-dir
	--with-stdint-include
	--without-stdint-include=${stdint-dir}/include
	--with-stdint-lib
	--without-stdint-lib=${stdint-dir}/lib
/usr/lib/ruby/2.7.0/mkmf.rb:471:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
	from /usr/lib/ruby/2.7.0/mkmf.rb:613:in `try_cpp'
	from /usr/lib/ruby/2.7.0/mkmf.rb:1124:in `block in have_header'
	from /usr/lib/ruby/2.7.0/mkmf.rb:971:in `block in checking_for'
	from /usr/lib/ruby/2.7.0/mkmf.rb:361:in `block (2 levels) in postpone'
	from /usr/lib/ruby/2.7.0/mkmf.rb:331:in `open'
	from /usr/lib/ruby/2.7.0/mkmf.rb:361:in `block in postpone'
	from /usr/lib/ruby/2.7.0/mkmf.rb:331:in `open'
	from /usr/lib/ruby/2.7.0/mkmf.rb:357:in `postpone'
	from /usr/lib/ruby/2.7.0/mkmf.rb:970:in `checking_for'
	from /usr/lib/ruby/2.7.0/mkmf.rb:1123:in `have_header'
	from extconf.rb:3:in `<main>'
rake aborted!
Command failed with status (1): [/usr/bin/ruby2.7 -S extconf.rb...]
/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/gems/digest-crc-0.6.1/ext/digest/Rakefile:32:in `block (3 levels) in <top
(required)>'
/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/gems/digest-crc-0.6.1/ext/digest/Rakefile:31:in `chdir'
/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/gems/digest-crc-0.6.1/ext/digest/Rakefile:31:in `block (2 levels) in <top
(required)>'
/usr/share/rubygems-integration/all/gems/rake-13.0.1/exe/rake:27:in `<main>'
Tasks: TOP => default => crc5/crc5_ext.so => crc5/Makefile
(See full trace by running task with --trace)

rake failed, exit code 1

Gem files will remain installed in /tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/gems/digest-crc-0.6.1 for inspection.
Results logged to
/tmp/bundler20201007-36133-1fk7e5pdigest-crc-0.6.1/extensions/x86_64-linux/2.7.0/digest-crc-0.6.1/gem_make.out

An error occurred while installing digest-crc (0.6.1), and Bundler cannot continue.
Make sure that `gem install digest-crc -v '0.6.1' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  fastlane was resolved to 2.162.0, which depends on
    google-cloud-storage was resolved to 1.29.1, which depends on
      digest-crc


```
I have also tried `sudo gem install fastlane -NV` but I get the same error.

I discovered #16725 and tried the [solution](https://github.com/fastlane/fastlane/issues/16725#issuecomment-652778837) mentioned, it didn't work for me. I also tried the [second](https://github.com/fastlane/fastlane/issues/16725#issuecomment-670398496) solution mentioned on the issue, that didn't work either because I'm already using Ruby 2.7.0.

In conclusion, I have tried both ways to install fastlane in Linux but none of those worked. Please enlighten me.

##### Command executed

`sudo bundle update`

##### Complete output when running fastlane, including the stack trace and command used
Can't run fastlane because it's not installed yet.

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment
I'm using Ubuntu 20.04.1
I'm using Ruby 2.7.0

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by iadeelzafar** [comment]
@iur-kaia @joshdholtz @rogerluan @kishikawakatsumi

**Comment by iur-kaia** [comment]
@iadeelzafar 
Apparently my issue was related to a specific VM image on CircleCI. The ruby version there was somehow broken. What I had to do was to change the ruby version to >2.7.0 on that image. CircleCI since then also has updated the image so I don't encounter the issue anymore.

**Comment by iadeelzafar** [comment]
So the primary issue was ```/usr/lib/ruby/2.7.0/mkmf.rb:471:in `try_do': The compiler failed to generate an executable file. (RuntimeError)``` as you can see in the logs.

Fixed it after installing [these](https://github.com/travis-ci/travis.rb/issues/558#issuecomment-355962361) dependencies and also `sudo apt-get install build-essential g++`

**Comment by JaminZhou** [comment]
If you are using a mac, you can try it.

1. sudo rm -rf /Library/Developer/CommandLineTools
2. xcode-select --install
3. sudo xcodebuild -license accept
4. open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg
